.clang-tidy

```diff
@@ -24,8 +24,29 @@ Checks: >
   -bugprone-switch-missing-default-case,
   -bugprone-unsafe-functions,
   -bugprone-unused-return-value,
+  -cert-dcl37-c,
+  -cert-dcl51-cpp,
+  -cert-env33-c,
+  -cert-err33-c,
+  -cert-err34-c,
+  -cert-exp42-c,
+  -cert-flp37-c,
+  -cert-msc24-c,
+  -cert-msc33-c,
+  -concurrency-mt-unsafe,
+  -cppcoreguidelines-avoid-magic-numbers,
+  -cppcoreguidelines-avoid-non-const-global-variables,
+  -cppcoreguidelines-init-variables,
+  -cppcoreguidelines-narrowing-conversions,
   -google-readability-braces-around-statements,
```

Comment:
i would expect we could enable this one

Reply:
It fails on `memline.c` on line 1276:

```c
  if (fname != NULL) {
#ifdef HAVE_READLINK
    // Expand symlink in the file name, because the swapfile is created
    // with the actual file instead of with the symlink.
    if (resolve_symlink(fname, fname_buf) == OK) {
      fname_res = fname_buf;
    } else
#endif  // <-- right here
    fname_res = fname;
  }
```

I guess we can disable the warning on that line only, but the check isn't really needed to begin with since uncrustify already adds braces around all statements. I don't mind either way, what do you prefer?

Reply:
I see. Is it helpful to eliminate as many from here as possible, so that it's easier to review the remaining ones?

Reply:
I think that's a great idea, it'll help with the bookkeeping the warnings.

Reply:
OK, @ii14 had a big brain suggestion for this so we can remove the braces lines.